x-pack/plugins/infra/server/lib/adapters/nodes/extract_paths_and_metrics.ts

```diff
@@ -77,8 +77,18 @@ function extractArgument<ReturnType>(selection: FieldNode, name: string): Return
 function extractPathArgument(selection: FieldNode): InfraPathInput[] {
   return extractArgument<InfraPathInput>(selection, 'path') as InfraPathInput[];
 }
-function extractMetricsArgument(selection: FieldNode): InfraMetricInput[] {
-  return extractArgument<InfraMetricInput>(selection, 'metrics') as InfraMetricInput[];
+
+function extractMetricArgument(selection: FieldNode): InfraMetricInput | null {
+  if (!selection.arguments) {
+    return null;
+  }
+  return selection.arguments
+    .filter(isArgumentNode)
+    .filter(subject => subject.name.value === 'metric')
+    .reduce((prev: InfraMetricInput[], argument: ArgumentNode) => {
+      return valueFromASTUntyped(argument.value);
+    }, [])
+    .pop();
```

Comment:
Does the above `reduce` return an array even though the `metric` argument is not an array anymore?